### PR TITLE
[numpy] add new ffi for column_stack and hstack

### PR DESCRIPTION
**Pull request by vexilligera**
## Description ##
[numpy] add new ffi for column_stack and hstack

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/operator/nn/concat-inl.h [hunk 1]
```diff
@@ -55,6 +55,14 @@ struct ConcatParam : public dmlc::Parameter<ConcatParam> {
     DMLC_DECLARE_FIELD(dim).set_default(1)
     .describe("the dimension to be concated.");
   }
+
+  void SetAttrDict(std::unordered_map<std::string, std::string>* dict) {
+    std::ostringstream ss;
+    ss << num_args;
+    (*dict)["num_args"] = ss.str();
+    ss << dim;
```
Comment (hzfan) [inline src/operator/nn/concat-inl.h]: Please clear `ss` before it is used for the second param, or use two `ostringstream` seperately.

**Comment by vexilligera** [comment]
@mxnet-bot run ci [centos-cpu]

**Comment by vexilligera** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by vexilligera** [comment]
@haojin2

**Comment by vexilligera** [comment]
@mxnet-bot run ci [unix-cpu]

**Comment by vexilligera** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by vexilligera** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by vexilligera** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by vexilligera** [comment]
@haojin2

**Pull request merged**